tests/Actual/CollectionUtilityTest.php

```diff
@@ -396,7 +399,7 @@ public function testThatCollectionCanBeSortedByNestedProperty()
     {
         $collection = [
             ["string" => "a", "nested" => ["number" => 1]],
-            ["string" => "d", "nested" => ["number" => 3]],
+            ["string" => "d", "nested" => ["number" => 4]],
```

Comment:
I've read the description and understand that there is some underlying php8+ change, but I don't understand the need to change the test input. Surely the input where there are two numbers the same is valid and should still be accounted for?

Reply:
The order changes. In PHP 7.4 the order of items with the same number is different to the order of numbers in PHP 8 for the reasons provided. You can try revert the test change yourself and run the tests to confirm.

Reply:
Yep, I understand that. I just think that we should still be testing for a scenario where the number is the same, even if it means that the output is different. With this update, we are loosing a test for a valid scenario. Or I might be completely misunderstanding what you mean.

Reply:
The output is not guarranteed in 7.4 and 8 so the output is different.

If we are supporting php 7.4 and php 8 with this package then this test CANNOT pass because php 7.4 and php 8 literally order the entries differently.

Again, i would suggest you revert the change push it up and watch the test cases play through.

Reply:
Now I understand! Sorry for being think 🤦‍♀️ 
Maybe we can just do what we did with the other packages? Drop php 7.4 support and have only php 8?

Reply:
I'm trying to see if i can avoid that as much as possible so the transition is less dangerous.